test_communication/test/test_subscriber.cpp

```diff
@@ -141,7 +139,14 @@ int main(int argc, char ** argv)
     rclcpp::shutdown();
     return 1;
   }
-  rclcpp::spin(node);
+
+  try {
+    rclcpp::spin(node);
+  } catch (const std::exception & e) {
+    std::cerr << e.what() << std::endl;
```

Comment:
Please use `fprintf` here (see https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#c-specific).

Reply:
Uh, how's that `fprintf` prevents `stderr` interleaving?

Reply:
Both operators are executed separately and have a chance of being interleaved by any other concurrent write to `stdout`. The `printf` creates a single string including the newline and writes them in a single call to `stdout`. While the latter technicall doesn't guarantee the operation to be atomic it practically makes it much much less likely to be interleaved.

Reply:
Thanks for the explanation, fixed in bbc82e9.